logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -58,7 +61,7 @@ def close
       end
 
       def empty?
-        true # synchronous queue is alway empty
+        @queue.queue.isEmpty
```

Comment:
@andrewvc nah careful, we're in `ReadClient` here and `WrappedSynchronousQueue` doesn't define an `isEmpty?` method :)